Makefile

```diff
@@ -67,4 +67,14 @@ distclean:
 	rm -rf tmp_install
 	cargo clean
 
+fmt:
+	@files=$$(git diff --cached --name-only --diff-filter=ACM | grep ".rs"); \
+	if [ "$$files" ]; then \
+		rustfmt $$files --edition=2018; \
+	fi;
+
+
+setup-pre-commit-hook:
```

Comment:
Another phony target. Maybe it would be better to split one big `.PHONY` directive into smaller ones, each above the corresponding target.

Reply:
done